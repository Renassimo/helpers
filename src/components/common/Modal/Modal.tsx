import { ReactNode } from 'react';

import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';
import LoadingButton from '@mui/lab/LoadingButton';
import IconButton from '@mui/material/IconButton';
import CloseIcon from '@mui/icons-material/Close';

import useMediaQuery from '@mui/material/useMediaQuery';
import { useTheme } from '@mui/material/styles';

const Modal = ({
  open,
  children,
  onClose,
  onSubmit,
  title,
  loading,
}: {
  open: boolean;
  children: ReactNode;
  onClose: () => void;
  onSubmit: () => void;
  title: string;
  loading: boolean;
}) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
  return (
    <Dialog
      fullScreen={fullScreen}
      open={open}
      onClose={onClose}
      aria-labelledby="responsive-modal"
    >
      <DialogTitle id="responsive-modal">
        {title}
        {onClose ? (
          <IconButton
            aria-label="close"
            onClick={onClose}
            sx={{
              position: 'absolute',
              right: 8,
              top: 8,
              color: (theme) => theme.palette.grey[500],
            }}
          >
            <CloseIcon />
          </IconButton>
        ) : null}
      </DialogTitle>
      <DialogContent>{children}</DialogContent>
      <DialogActions>
        <LoadingButton onClick={onSubmit} loading={loading} autoFocus>
          Save
        </LoadingButton>
      </DialogActions>
    </Dialog>
  );
};

export default Modal;
