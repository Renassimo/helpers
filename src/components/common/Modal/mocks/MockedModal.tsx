import { FormEvent, ReactNode } from 'react';
import { Breakpoint } from '@mui/material';

const MockedModal = jest.fn(
  ({
    open,
    children,
    onClose,
    onSubmit,
    title,
    loading,
    maxWidth = 'sm',
  }: {
    open: boolean;
    children: ReactNode;
    onClose: () => void;
    onSubmit: (event: FormEvent<HTMLFormElement>) => void;
    title: string;
    loading: boolean;
    maxWidth?: Breakpoint;
  }) => (
    <div>
      {children}
      <div>
        title: {title}; maxWidth: {maxWidth};{open ? 'open' : 'closed'};
        {loading ? 'loading' : 'not loading'}; onClose{' '}
        {!!onClose ? 'passed' : 'did not passed'}; onSubmit{' '}
        {!!onSubmit ? 'passed' : 'did not passed'};
      </div>
    </div>
  )
);

export default MockedModal;
