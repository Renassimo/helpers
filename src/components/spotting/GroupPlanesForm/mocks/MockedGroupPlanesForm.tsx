const MockedGroupPlanesForm = jest.fn(() => <div>MockedGroupPlanesForm</div>);

export default MockedGroupPlanesForm;
