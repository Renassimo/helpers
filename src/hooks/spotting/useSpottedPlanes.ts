import { useContext } from 'react';

import SpottingContext from '@/contexts/spotting';

const useSpottedPlanes = () => useContext(SpottingContext);

export default useSpottedPlanes;
