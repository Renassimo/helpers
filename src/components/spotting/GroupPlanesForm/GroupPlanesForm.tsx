import React, { ChangeEvent, useMemo } from 'react';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';

import { useTheme } from '@mui/material/styles';

import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

const HASHTAG_LIMIT = 30;

const GroupPlanesForm = () => {
  const {
    groupDescription,
    groupHashtags,
    groupName,
    setGroupDescription,
    setGroupHashtags,
    setGroupName,
  } = useSpottedPlanes();

  const descriptionInputName = `group-description`;
  const hashtagsInputName = `group-hashtags`;
  const groupNameInputName = `group-name`;

  const onDescriptionChange = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    setGroupDescription(event.target.value);
  };
  const onHashtagsChange = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    setGroupHashtags(event.target.value);
  };
  const onGroupNameChange = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    setGroupName(event.target.value);
  };

  const theme = useTheme();
  const descriptionHashtagsCount = useMemo(
    () => groupDescription.split('#').length - 1,
    [groupDescription]
  );
  const hashtagsCount = useMemo(
    () => groupHashtags.split('#').length - 1,
    [groupHashtags]
  );
  const totalHashTags = useMemo(
    () => descriptionHashtagsCount + hashtagsCount,
    [descriptionHashtagsCount, hashtagsCount]
  );
  const hashtagsExceeded = useMemo(
    () => totalHashTags > HASHTAG_LIMIT,
    [totalHashTags]
  );
  const hashtagColor = useMemo(
    () =>
      hashtagsExceeded
        ? theme.palette.warning.main
        : theme.palette.primary.main,
    [hashtagsExceeded, theme.palette.primary]
  );

  return (
    <>
      <TextField
        id={descriptionInputName}
        name={descriptionInputName}
        value={groupDescription}
        onChange={onDescriptionChange}
        multiline
        fullWidth
        margin="dense"
        variant="standard"
        placeholder="Description"
      />
      <Typography variant="caption" color={hashtagColor}>
        Hashtags: {descriptionHashtagsCount}
      </Typography>
      <TextField
        id={hashtagsInputName}
        name={hashtagsInputName}
        value={groupHashtags}
        onChange={onHashtagsChange}
        multiline
        fullWidth
        margin="dense"
        variant="standard"
        placeholder="Hashtags"
      />
      <Typography variant="caption" color={hashtagColor}>
        Hashtags: {hashtagsCount}
      </Typography>
      <Box display="flex" justifyContent="flex-end">
        <TextField
          id={groupNameInputName}
          name={groupNameInputName}
          value={groupName}
          onChange={onGroupNameChange}
          margin="dense"
          variant="standard"
          placeholder="Group name"
        />
      </Box>
    </>
  );
};

export default GroupPlanesForm;
