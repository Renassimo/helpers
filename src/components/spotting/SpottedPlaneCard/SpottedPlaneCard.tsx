import React from 'react';

import { SpottedPlaneProviderData } from '@/types/spotting';

import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import CardActions from '@mui/material/CardActions';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import Button from '@mui/material/Button';
import LoadingButton from '@mui/lab/LoadingButton';

import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import CollectionsIcon from '@mui/icons-material/Collections';
import ContentPasteIcon from '@mui/icons-material/ContentPaste';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';
import useApplySpottedPlanes from '@/hooks/spotting/useApplySpottedPlanes';

import SelectableCard from '@/components/spotting/SelectableCard';
import NewFirstFlightForm from '@/components/spotting/NewFirstFlightForm';
import SpottedPlaneForm from '@/components/spotting/SpottedPlaneForm';

const SpottedPlaneCard = ({
  data,
  selectable = false,
}: {
  data: SpottedPlaneProviderData;
  selectable?: boolean;
}) => {
  const {
    id,
    name,
    photoUrl,
    planespottersUrl,
    description,
    hashtags,
    firstFlight,
    photosUrl,
    url,
  } = data;

  const { update, loading } = useApplySpottedPlanes();

  const {
    selectedIds,
    addSelectedId,
    removeSelectedIds,
    generateDescription,
    generateHashtags,
    clearHashtags,
    clearDescription,
  } = useSpottedPlanes();
  const isAnySelected = selectedIds.length > 0;
  const isSelected = selectedIds.includes(id);

  const toggleSelect = () => {
    if (isSelected) {
      removeSelectedIds([id]);
    } else {
      addSelectedId(id);
    }
  };

  const isPhotoSelectable = selectable && isAnySelected;

  const onPhotoClick = () => {
    if (isPhotoSelectable) toggleSelect();
  };

  const createDescription = () => {
    generateDescription(id);
    generateHashtags(id);
  };

  const apply = async () => {
    await update([data]);
  };

  const discard = () => {
    clearDescription(id);
    clearHashtags(id);
  };

  const showDescription = !!(description || hashtags);

  return (
    <SelectableCard
      isAnySelected={isAnySelected}
      selectable={selectable}
      selected={isSelected}
      toggleSelect={toggleSelect}
    >
      {
        <CardMedia
          image={photoUrl ?? '/images/plane.png'}
          title={name ?? id}
          sx={{ height: 140, cursor: isPhotoSelectable ? 'pointer' : '' }}
          onClick={onPhotoClick}
        />
      }
      <CardContent>
        <Typography gutterBottom variant="h6" component="h2">
          {name}
          <IconButton size="small" href={url} target="_blank" rel="noreferrer">
            <ContentPasteIcon fontSize="small" />
          </IconButton>
          {photosUrl && (
            <IconButton
              size="small"
              href={photosUrl}
              target="_blank"
              rel="noreferrer"
            >
              <CollectionsIcon fontSize="small" />
            </IconButton>
          )}
          {planespottersUrl && (
            <IconButton
              size="small"
              href={planespottersUrl}
              target="_blank"
              rel="noreferrer"
            >
              <OpenInNewIcon fontSize="small" />
            </IconButton>
          )}
        </Typography>
        {!firstFlight && (
          <NewFirstFlightForm id={data.id} disabled={showDescription} />
        )}
        {showDescription && <SpottedPlaneForm data={data} />}
      </CardContent>
      <CardActions sx={{ justifyContent: 'end' }}>
        {showDescription ? (
          <>
            <Button size="small" onClick={discard} disabled={loading}>
              Discard
            </Button>
            {selectable && (
              <LoadingButton size="small" onClick={apply} loading={loading}>
                Apply
              </LoadingButton>
            )}
          </>
        ) : (
          <Button size="small" onClick={createDescription}>
            Create description
          </Button>
        )}
      </CardActions>
    </SelectableCard>
  );
};

export default SpottedPlaneCard;
