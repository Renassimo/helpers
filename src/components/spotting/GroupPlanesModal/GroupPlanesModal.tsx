import React, { useCallback, useEffect, useState } from 'react';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';
import useApplySpottedPlanes from '@/hooks/spotting/useApplySpottedPlanes';
import useAlerts from '@/hooks/alerts';

import Button from '@mui/material/Button';

import Modal from '@/components/common/Modal';
import SpottedPlaneCard from '@/components/spotting/SpottedPlaneCard';
import GroupPlanesForm from '@/components/spotting/GroupPlanesForm';
import PlanesContainer from '@/components/spotting/PlanesContainer';

const GroupPlanesModal = () => {
  const {
    spottedPlanes,
    selectedIds,
    generateGroupDescriptionAndHashtags,
    clearGroupData,
    clearSelectedIds,
    groupName,
    groupDescription,
    groupHashtags,
  } = useSpottedPlanes();
  const { update, loading } = useApplySpottedPlanes();
  const { createInfoAlert, clearAll } = useAlerts();

  const [isModalOpen, setIsModalOpen] = useState(false);

  const onCloseModal = useCallback(() => {
    setIsModalOpen(false);
    clearGroupData();
  }, [clearGroupData, setIsModalOpen]);

  const onOpenModal = useCallback(() => {
    generateGroupDescriptionAndHashtags();
    setIsModalOpen(true);
  }, [generateGroupDescriptionAndHashtags, setIsModalOpen]);

  const onSubmitModal = async () => {
    const data = spottedPlanes
      .filter((spottedPlane) => selectedIds.includes(spottedPlane.id))
      .map((spottedPlane) => ({
        ...spottedPlane,
        groupName,
        groupDescription,
        groupHashtags,
      }));
    await update(data);
    clearSelectedIds();
    onCloseModal();
  };

  useEffect(() => {
    const selectedCount = selectedIds.length;
    clearAll();
    if (selectedCount > 1)
      createInfoAlert(
        <span>
          Create group post from {selectedCount} planes?{' '}
          <Button type="button" onClick={onOpenModal}>
            Create
          </Button>
          <Button type="button" onClick={clearSelectedIds}>
            Cancel
          </Button>
        </span>,
        0
      );
  }, [clearAll, createInfoAlert, onOpenModal, selectedIds, clearSelectedIds]);

  return (
    <>
      <Modal
        open={isModalOpen}
        onClose={onCloseModal}
        onSubmit={onSubmitModal}
        title="Create group post"
        loading={loading}
        maxWidth={'xl'}
      >
        <GroupPlanesForm />
        <PlanesContainer>
          {spottedPlanes
            .filter((item) => selectedIds.includes(item.id))
            .map((item) => (
              <SpottedPlaneCard data={item} key={item.id} />
            ))}
        </PlanesContainer>
      </Modal>
    </>
  );
};

export default GroupPlanesModal;
