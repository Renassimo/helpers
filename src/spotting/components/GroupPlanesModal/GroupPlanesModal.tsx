import React, { Dispatch, SetStateAction, useCallback, useEffect } from 'react';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';
import useApplySpottedPlanes from '@/hooks/spotting/useApplySpottedPlanes';
import useAlerts from '@/hooks/alerts';

import Modal from '@/common/components/Modal';
import SpottedPlaneCard from '@/spotting/components/SpottedPlaneCard';
import GroupPlanesForm from '@/spotting/components/GroupPlanesForm';
import PlanesContainer from '@/spotting/components/PlanesContainer';
import GroupPostInfoAlert from '@/spotting/components/GroupPostInfoAlert';

const GroupPlanesModal = ({
  isModalOpen,
  setIsModalOpen,
}: {
  isModalOpen: boolean;
  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
}) => {
  const {
    spottedPlanes,
    selectedIds,
    generateGroupDescriptionAndHashtags,
    clearGroupData,
    clearSelectedIds,
    groupName,
    groupDescription,
    groupHashtags,
  } = useSpottedPlanes();
  const { update, loading } = useApplySpottedPlanes();
  const { createInfoAlert, clearAll } = useAlerts();

  const onCloseModal = useCallback(() => {
    setIsModalOpen(false);
    clearGroupData();
  }, [clearGroupData, setIsModalOpen]);

  const onOpenModal = useCallback(() => {
    generateGroupDescriptionAndHashtags();
    setIsModalOpen(true);
  }, [generateGroupDescriptionAndHashtags, setIsModalOpen]);

  const onSubmitModal = async () => {
    const data = spottedPlanes
      .filter((spottedPlane) => selectedIds.includes(spottedPlane.id))
      .map((spottedPlane) => ({
        ...spottedPlane,
        groupName,
        groupDescription,
        groupHashtags,
      }));
    await update(data);
    clearSelectedIds();
    onCloseModal();
  };

  useEffect(() => {
    clearAll();

    const selectedCount = selectedIds.length;

    if (selectedCount > 1)
      createInfoAlert(
        <GroupPostInfoAlert
          selectedCount={selectedCount}
          onOpenModal={onOpenModal}
          clearSelectedIds={clearSelectedIds}
        />,
        0
      );
  }, [clearAll, createInfoAlert, onOpenModal, selectedIds, clearSelectedIds]);

  return (
    <>
      <Modal
        open={isModalOpen}
        onClose={onCloseModal}
        onSubmit={onSubmitModal}
        title="Create group post"
        loading={loading}
        disabled={!groupName}
        maxWidth={'xl'}
      >
        <GroupPlanesForm />
        <PlanesContainer>
          {spottedPlanes
            .filter((item) => selectedIds.includes(item.id))
            .map((item) => (
              <SpottedPlaneCard data={item} key={item.id} />
            ))}
        </PlanesContainer>
      </Modal>
    </>
  );
};

export default GroupPlanesModal;
