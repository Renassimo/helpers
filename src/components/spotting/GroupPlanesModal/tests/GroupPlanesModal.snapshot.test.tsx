import renderWithTheme from '@/tests/helpers';

import useSpottedPlanes from '@/hooks/spotting/useSpottedPlanes';
import useApplySpottedPlanes from '@/hooks/spotting/useApplySpottedPlanes';

import Modal from '@/common/components/Modal';
import GroupPlanesModal from '@/components/spotting/GroupPlanesModal';
import GroupPlanesForm from '@/components/spotting/GroupPlanesForm';
import SpottedPlaneCard from '@/components/spotting/SpottedPlaneCard';

import MockedModal from '@/common/components/Modal/mocks';
import MockedGroupPlanesForm from '@/components/spotting/GroupPlanesForm/mocks';
import MockedSpottedPlaneCard from '@/components/spotting/SpottedPlaneCard/mocks';

import {
  mockedSpottedPlaneProviderDataFalsy,
  mockedSpottedPlaneProviderDataMixed,
  mockedSpottedPlaneProviderDataTruthy,
} from '@/types/spotting/mocks/mockedSpottedPlaneProviderData';

jest.mock('@/common/components/Modal');
jest.mock('@/hooks/spotting/useSpottedPlanes');
jest.mock('@/hooks/spotting/useApplySpottedPlanes');
jest.mock('@/components/spotting/GroupPlanesForm');
jest.mock('@/components/spotting/SpottedPlaneCard');

describe('GroupPlanesModal snapshot', () => {
  let loading = false;

  const mockedSetIsModalOpen = jest.fn();

  const mockedSpottedPlanes = [
    mockedSpottedPlaneProviderDataTruthy,
    mockedSpottedPlaneProviderDataMixed,
    mockedSpottedPlaneProviderDataFalsy,
  ];

  const mockedSelectedIds = [
    mockedSpottedPlaneProviderDataTruthy.id,
    mockedSpottedPlaneProviderDataFalsy.id,
  ];

  beforeEach(() => {
    (useSpottedPlanes as unknown as jest.Mock).mockImplementation(() => ({
      spottedPlanes: mockedSpottedPlanes,
      selectedIds: mockedSelectedIds,
      generateGroupDescriptionAndHashtags: jest.fn(),
      groupName: 'groupName',
      clearGroupData: jest.fn(),
      clearSelectedIds: jest.fn(),
    }));
    (useApplySpottedPlanes as unknown as jest.Mock).mockImplementation(() => ({
      loading,
    }));
    (SpottedPlaneCard as unknown as jest.Mock).mockImplementation(
      MockedSpottedPlaneCard
    );
    (Modal as unknown as jest.Mock).mockImplementation(MockedModal);
    (GroupPlanesForm as unknown as jest.Mock).mockImplementation(
      MockedGroupPlanesForm
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arrange
    // Act
    const { baseElement } = renderWithTheme(
      <GroupPlanesModal isModalOpen setIsModalOpen={mockedSetIsModalOpen} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when loading', () => {
    test('renders successfully', () => {
      // Arrange
      loading = true;
      // Act
      const { baseElement } = renderWithTheme(
        <GroupPlanesModal isModalOpen setIsModalOpen={mockedSetIsModalOpen} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('when no group name', () => {
    beforeEach(() => {
      (useSpottedPlanes as unknown as jest.Mock).mockImplementation(() => ({
        spottedPlanes: mockedSpottedPlanes,
        selectedIds: mockedSelectedIds,
        generateGroupDescriptionAndHashtags: jest.fn(),
        clearGroupData: jest.fn(),
        clearSelectedIds: jest.fn(),
        groupName: '',
      }));
    });

    test('renders successfully', () => {
      // Arrange
      loading = false;
      // Act
      const { baseElement } = renderWithTheme(
        <GroupPlanesModal isModalOpen setIsModalOpen={mockedSetIsModalOpen} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
