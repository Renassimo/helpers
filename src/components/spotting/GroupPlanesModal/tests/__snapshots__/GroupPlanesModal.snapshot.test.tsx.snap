// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupPlanesModal snapshot renders successfully 1`] = `
<body>
  <div>
    <div>
      <div>
        MockedGroupPlanesForm
      </div>
      <div
        class="css-zsqo3r"
      >
        <div>
          Mocked Spotted Plane Card:
           
          not selectable
          , id: 
          id-truthy
          , name:
          Qatar Airways A350-900 A7-ALB
          , photoUrl; 
          https://file.notion.so/
          , planespottersUrl: 
          https://www.planespotters.net/
          , description: 
          description 1
          , hashtags: 
          #hashtags1
        </div>
        <div>
          Mocked Spotted Plane Card:
           
          not selectable
          , id: 
          id-falsy
          , name:
          LOT 787-8 SP-LRF
          , photoUrl; 
          https://file.notion.so/
          , planespottersUrl: 
          https://www.planespotters.net/
          , description: 
          description 2
          , hashtags: 
          #hashtags2
        </div>
      </div>
      <div>
        title: 
        Create group post
        ; maxWidth: 
        xl
        ;
        open
        ;
        not loading
        ; onClose
         
        passed
        ; onSubmit
         
        passed
        ;
      </div>
    </div>
  </div>
</body>
`;

exports[`GroupPlanesModal snapshot when loading renders successfully 1`] = `
<body>
  <div>
    <div>
      <div>
        MockedGroupPlanesForm
      </div>
      <div
        class="css-zsqo3r"
      >
        <div>
          Mocked Spotted Plane Card:
           
          not selectable
          , id: 
          id-truthy
          , name:
          Qatar Airways A350-900 A7-ALB
          , photoUrl; 
          https://file.notion.so/
          , planespottersUrl: 
          https://www.planespotters.net/
          , description: 
          description 1
          , hashtags: 
          #hashtags1
        </div>
        <div>
          Mocked Spotted Plane Card:
           
          not selectable
          , id: 
          id-falsy
          , name:
          LOT 787-8 SP-LRF
          , photoUrl; 
          https://file.notion.so/
          , planespottersUrl: 
          https://www.planespotters.net/
          , description: 
          description 2
          , hashtags: 
          #hashtags2
        </div>
      </div>
      <div>
        title: 
        Create group post
        ; maxWidth: 
        xl
        ;
        open
        ;
        loading
        ; onClose
         
        passed
        ; onSubmit
         
        passed
        ;
      </div>
    </div>
  </div>
</body>
`;
